pinot-common/src/main/java/com/linkedin/pinot/common/data/StarTreeIndexSpec.java

```diff
@@ -40,9 +45,25 @@
 
   private boolean _excludeSkipMaterializationDimensionsForStarTreeIndex;
 
+  private static final String SEPARATOR = ",";
+  private static final String DIMENSIONS_SPLIT_ORDER_PARAM = "dimensionsSplitOrder";
+  private static final String MAX_LEAF_RECORDS_PARAM = "maxLeafRecords";
+  private static final String SKIP_MATERIALIZATION_CARDINALITY_THRESHOLD_PARAM = "skipMaterializationCardinalityThreshold";
+  private static final String SKIP_MATERIALIZATION_FOR_DIMENSIONS_PARAM = "skipMaterializationForDimensions";
+  private static final String SKIP_STAR_NODE_CREATION_FOR_DIMENSIONS_PARAM = "skipStarNodeCreationForDimensions";
+
   public StarTreeIndexSpec() {
   }
 
+  public StarTreeIndexSpec(String jsonString) {
```

Comment:
Could we use a static method fromJson() for this, instead of a separate constructor?

Reply:
done